replication.go

```diff
@@ -0,0 +1,201 @@
+// Copyright (C) 2019-2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package simplex
+
+import (
+	"bytes"
+
+	"go.uber.org/zap"
+)
+
+type Request struct {
+	FinalizationCertificateRequest *FinalizationCertificateRequest
+	LatestRoundRequest             *LatestRoundRequest
+}
+
+type Response struct {
+	FinalizationCertificateResponse *FinalizationCertificateResponse
+	LatestRoundResponse             *LatestRoundResponse
+}
+
+// request a finalization certificate for the given sequence number
+type FinalizationCertificateRequest struct {
+	Seq uint64
+}
+
+type FinalizationCertificateResponse struct {
+	FCert FinalizationCertificate
+	Block Block
+}
+
+type LatestRoundRequest struct {
+}
+
+// if block has not been proposed yet, we should return the last round's block, notarization, and fCert
+type LatestRoundResponse struct {
```

Comment:
Maybe it would be more succinct to just send the notarization and fCert without the block? And then the requester would have the freedom to request the block if it needs it, just to reduce the network bandwidth.